### ISSUE TITLE
Borgbackup timer doesn't depend on network-online.target

### ISSUE THREAD

**Issue by nielsegberts** [NixOS/nixpkgs#282640]
### Describe the bug

On unstable, Borgbackup jobs created with `persistentTimer` fail to build because they don't depend on network-online.target.

### Steps To Reproduce

When adding a borgbackup job to configuration.nix with `persistentTimer`

``` 
 services.borgbackup.jobs.home-danbst = {
    paths = "/home/danbst";
    encryption.mode = "none";
    environment.BORG_RSH = "ssh -i /home/danbst/.ssh/id_ed25519";
    repo = "ssh://user@example.com:23/path/to/backups-dir/home-danbst";
    compression = "auto,zstd";
    startAt = "daily";
    persistentTimer = true;
  };
```

It throws the following error on `nixos-rebuild switch`:

```
borgbackup-job-home-danbst.timer is ordered after 'network-online.target' but doesn't depend on it
```

### Additional context

I think this bug was introduced when this PR was merged:

https://github.com/NixOS/nixpkgs/pull/258680

The solution might be as simple as adding `after = [ "network-online.target" ];` when `persistentTimer` is true here?

https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/backup/borgbackup.nix#L125

### Notify maintainers

@globin 
@dotlambda

**Comment by lf-** [comment]
Oh yeah, that's a dynamically generated config that got missed by our grepping, and is not tested in a NixOS test, so it simply slipped through the cracks.

**Comment by Stunkymonkey** [comment]
i guess this is fixed via #301528?

**Comment by bjornfor** [comment]
> i guess this is fixed via #301528?

Seems so.
